lib/providers/portion/PortionProvider.ts

```diff
@@ -39,6 +38,33 @@ export class PortionProvider implements IPortionProvider {
     return tokenOutCurrencyAmount.multiply(new Fraction(portion.bips, 10000));
   }
 
+  getPortionQuoteAmount(
+    portionAmountToken: CurrencyAmount<Currency>,
+    quoteCurrencyAmount: CurrencyAmount<Currency>,
+    amount: CurrencyAmount<Currency>
+  ): CurrencyAmount<Currency> {
+    // 1. we know the portion amount for exact out with 100% correctness,
+    //    so we can add the portion amount into the exact out amount swapper requested.
+    //    i.e. portionAdjustedAmount = amount + portionAmountToken
+    const portionAdjustedAmount = amount.add(portionAmountToken);
+    // 2. then we know portion amount and portion adjusted exact out amount,
+    //    we can get a ratio
+    //    i.e. portionToPortionAdjustedAmountRatio = portionAmountToken / portionAdjustedAmount
+    const portionToPortionAdjustedAmountRatio = new Fraction(
```

Comment:
I think I understand what `portionToPortionAdjustedAmountRatio` is referencing but the naming is a little confusing. This is essentially the fee token spot price in terms of input token, right?